### PR TITLE
Bundle ijavascript

### PR DESCRIPTION
**Pull request by lgeiger**
Based on #1338 

This is mainly for demonstration purpose ~~and not tested with a packaged release~~:
ikernel

### CODE REVIEW

File: src/main/kernel-specs.js [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+
+export default function saveKernelspecs(kernelSpecs) {
+  global.KERNEL_SPECS = {
+    node: {
+      name: 'node',
+      spec: {
+        argv: [
+          process.execPath,
+          path.dirname(require.resolve('ikernel')),
+          '{connection_file}'
```
Comment (rgbkrk) [inline src/main/kernel-specs.js]: Super cool!

**Comment by rgbkrk** [comment]
Feedback from video chat this morning:

We should probably build on top of [`jp-kernel`](https://github.com/n-riesco/jp-kernel) for our embedded kernel or we'll have to build a lot more into `ikernel` (which was largely created as a demo js kernel by myself).

**Comment by lgeiger** [comment]
Against my assumptions `ijavascript` doesn't install the kernel spec on `npm install` which is awesome and makes it usable for bundling.

Let's get this PR out of the demo stage 🚀  🔜

File: package.json [hunk 2]
```diff
@@ -132,6 +132,7 @@
     "enchannel": "^1.1.3",
     "enchannel-zmq-backend": "^2.6.0",
     "github": "^8.0.0",
+    "ijavascript": "github:lgeiger/ijavascript",
```
Comment (lgeiger) [inline package.json]: Waiting for https://github.com/n-riesco/ijavascript/pull/92 to be merged.

**Comment by lgeiger** [comment]
Ready to try out:
ijavascript

**Comment by rgbkrk** [comment]
I'm a bit hesitant with bringing in something that relies on a github repo directly, want to publish a namespaced package of `ijavascript` as `@lgeiger/ijavascript`?

**Comment by lgeiger** [comment]
> I'm a bit hesitant with bringing in something that relies on a github repo directly

Me too, if think we can wait till https://github.com/n-riesco/ijavascript/pull/92 gets merged

**Comment by rgbkrk** [comment]
Should we label this as "node.js (nteract)" for the display name so it doesn't confust/conflict with an installed JavaScript kernel?

File: src/main/kernel-specs.js [hunk 3]
```diff
@@ -1,9 +1,29 @@
...
+import { join } from 'path';
 
-const KERNEL_SPECS = {};
+const KERNEL_SPECS = {
+  node_nteract: {
+    name: 'node_nteract',
+    spec: {
+      argv: [
+        process.execPath,
+        join(require.resolve('ijavascript'), '..', 'lib', 'kernel.js'),
```
Comment (lgeiger) [inline src/main/kernel-specs.js:10]: A bit hacky but it works as long as ijavascripts folder structure doesn't change.

For the long term we can make a custom kernel as outlined in https://github.com/n-riesco/ijavascript/pull/92.

Comment (rgbkrk) [reply src/main/kernel-specs.js:10]: If we pin exactly we'll be alright.

**Review by rgbkrk — APPROVED** [review]

**Comment by rgbkrk** [comment]
We should switch our status bar to showing the display name of the kernel I think:

screen shot 2017-01-18 at 9 39 17 am

**Comment by captainsafia** [comment]
> We should switch our status bar to showing the display name of the kernel I think:

That makes sense. I believe I used to spec name when I first made the status bar just because it was what we stored in the state tree, so no real reason to have it this way in any case.

**Pull request merged**